collectors/apps.plugin/apps_plugin.c

```diff
@@ -3736,562 +3746,250 @@ static void normalize_utilization(struct target *root) {
-    }
-    send_END();
-
-    send_BEGIN(type, "cpu_user", dt);
     for (w = root; w ; w = w->next) {
-        if(unlikely(w->exposed && w->processes))
-            send_SET(w->name, (kernel_uint_t)(w->utime * utime_fix_ratio) + (include_exited_childs?((kernel_uint_t)(w->cutime * cutime_fix_ratio)):0ULL));
-    }
-    send_END();
-
-    send_BEGIN(type, "cpu_system", dt);
-    for (w = root; w ; w = w->next) {
-        if(unlikely(w->exposed && w->processes))
-            send_SET(w->name, (kernel_uint_t)(w->stime * stime_fix_ratio) + (include_exited_childs?((kernel_uint_t)(w->cstime * cstime_fix_ratio)):0ULL));
-    }
-    send_END();
+        if(unlikely(!w->exposed))
+            continue;
 
-    if(show_guest_time) {
-        send_BEGIN(type, "cpu_guest", dt);
-        for (w = root; w ; w = w->next) {
-            if(unlikely(w->exposed && w->processes))
-                send_SET(w->name, (kernel_uint_t)(w->gtime * gtime_fix_ratio) + (include_exited_childs?((kernel_uint_t)(w->cgtime * cgtime_fix_ratio)):0ULL));
-        }
+        send_BEGIN(type, w->clean_name, "processes", dt);
+        send_SET("processes", w->processes);
         send_END();
-    }
 
-#ifndef __FreeBSD__
-    send_BEGIN(type, "voluntary_ctxt_switches", dt);
-    for (w = root; w ; w = w->next) {
-        if(unlikely(w->exposed && w->processes))
-            send_SET(w->name, w->status_voluntary_ctxt_switches);
-    }
-    send_END();
-
-    send_BEGIN(type, "involuntary_ctxt_switches", dt);
-    for (w = root; w ; w = w->next) {
-        if(unlikely(w->exposed && w->processes))
-            send_SET(w->name, w->status_nonvoluntary_ctxt_switches);
-    }
-    send_END();
-#endif
-
-    send_BEGIN(type, "threads", dt);
-    for (w = root; w ; w = w->next) {
-        if(unlikely(w->exposed))
-            send_SET(w->name, w->num_threads);
-    }
-    send_END();
-
-    send_BEGIN(type, "processes", dt);
-    for (w = root; w ; w = w->next) {
-        if(unlikely(w->exposed))
-            send_SET(w->name, w->processes);
-    }
-    send_END();
-
-#ifndef __FreeBSD__
-    send_BEGIN(type, "uptime", dt);
-    for (w = root; w ; w = w->next) {
-        if(unlikely(w->exposed && w->processes))
-            send_SET(w->name, (global_uptime > w->starttime)?(global_uptime - w->starttime):0);
-    }
-    send_END();
-
-    if (enable_detailed_uptime_charts) {
-        send_BEGIN(type, "uptime_min", dt);
-        for (w = root; w ; w = w->next) {
-            if(unlikely(w->exposed && w->processes))
-                send_SET(w->name, w->uptime_min);
-        }
+        send_BEGIN(type, w->clean_name, "threads", dt);
+        send_SET("threads", w->num_threads);
         send_END();
 
-        send_BEGIN(type, "uptime_avg", dt);
-        for (w = root; w ; w = w->next) {
-            if(unlikely(w->exposed && w->processes))
-                send_SET(w->name, w->uptime_sum / w->processes);
-        }
-        send_END();
+        if(unlikely(!w->processes))
+            continue;
```

Comment:
I'm not entirely sure how it works, but `w->processes` has been set a few lines before?

Reply:
What do you mean? The goal of this `if` is to no send metrics for empty groups (with 0 processes).